### ISSUE TITLE
Unity + LUIS hololens 2 tutorial out of date

### ISSUE THREAD

**Issue by yuchad** [MicrosoftDocs/mixed-reality#137]
Hi there, I have been attempting to implement Azure Speech services for Unity and the hololens 2.

I have been able to successfully implement step 1-3 fairly easily and quickly.

Unfortunately, step 4, setting up intent and natural language understanding, seems to be having quite a bit of issues. 

The issues can be described by a user at this forum post: https://docs.microsoft.com/en-us/answers/questions/110857/object-reference-not-set-to-an-instance-of-an-obje.html

Essentially, setting up the LUIS portal and intents is no problem. I can even verify that the endpoint utterances are coming through because I can see them on the portal!

However, it seems that the way the data is being parsed within the Unity application does not line up with how the LUIS endpoints are coming in.

It would be great if there was possibly an update, or a way to find out what's happening.

Thank you!

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 519e92d1-bd95-93c8-7e97-d1df949ed882
* Version Independent ID: bf701f2a-41dc-c37c-d947-28d3a4bcce73
* Content: [Azure Speech Services tutorials - 4. Setting up intent and natural language understanding - Mixed Reality](https://docs.microsoft.com/en-us/windows/mixed-reality/develop/unity/tutorials/mrlearning-speechsdk-ch4)
* Content Source: [mixed-reality-docs/mr-dev-docs/develop/unity/tutorials/mrlearning-speechSDK-ch4.md](https://github.com/MicrosoftDocs/mixed-reality/blob/docs/mixed-reality-docs/mr-dev-docs/develop/unity/tutorials/mrlearning-speechSDK-ch4.md)
* Product: **mixed-reality**
* Technology: **mr-dev**
* GitHub Login: @jessemcculloch
* Microsoft Alias: **jemccull**

**Comment by hferrone** [comment]
@martinwilter, @mr0ng or team - can you assist or is this a support question?

**Comment by yuchad** [comment]
@hferrone Appreciate you bringing this to others who would know about this!

Additionally, it seems that the forum post I sent you has to do with a hololens 1 project, however, comparing the source code, they are essentially the same with different function names as the tutorial provided in the hololens 2 tutorials. Ultimately, I have the same error, which I believe has to do with the way `SubmitRequestToLuis` function is parsing the results.

Any help would be greatly appreciated!

Thank you.

**Comment by martinwilter** [comment]
@hferrone, I haven't looked into the new LUIS portal yet, so not sure if this issue might be caused by those changes.

@yuchad, are you seeing any error messages that you can share?

**Comment by yuchad** [comment]
Hi @martinwilter 

```
Luis Request Exception Message: Object reference not set to an instance of an object
UnityEngine.Debug:Log(Object)
<SubmitRequestToLuis>d__18:MoveNext() (at Assets/MRTK.Tutorials.AzureSpeechServices/Scripts/LunarcomIntentRecognizer.cs:159)
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr) (at C:/buildslave/unity/build/Runtime/Export/Scripting/Coroutines.cs:17)
```

And doing some debugging, it's possible that on line 151, 

` AnalysedQuery analysedQuery = JsonUtility.FromJson<AnalysedQuery>(unityWebRequest.downloadHandler.text);`

the analysedQuery is perhaps not being properly parsed.

I try to run a ` Debug.Log(analysedQuery.topScoringIntent.intent);` but, I am simply getting `Null` which leads me to believe that possibility the JsonUtility isn't properly moving the strings to the analysedQuery class.

Please let me know if there's anything I can help with!

**Comment by martinwilter** [comment]
Hi @yuchad, thanks for the info.

I think you are correct, the culprit is on line 151 when creating the AnalysedQuery object from the JSON. Looks like this is caused by LUIS upgrade to v3 and changes to the response structure and variable names, for example, topScoringIntent is now topIntent.

To work around this you would have to recreate the object (AnalysedQuery and probably also TopScoringIntentData and EntityData) to match the JSON representation and also update the UnpackResults() function accordingly.

In summary, this tutorial is outdated and no longer supports the current LUIS portal.

I recommend you follow issue https://github.com/MicrosoftDocs/mixed-reality/issues/51 to be notified if/when the tutorial is updated.
__


CC @hferrone

**Comment by yuchad** [comment]
> Hi @yuchad, thanks for the info.
> 
> I think you are correct, the culprit is on line 151 when creating the AnalysedQuery object from the JSON. Looks like this is caused by LUIS upgrade to v3 and changes to the response structure and variable names, for example, topScoringIntent is now topIntent.
> 
> To work around this you would have to recreate the object (AnalysedQuery and probably also TopScoringIntentData and EntityData) to match the JSON representation and also update the UnpackResults() function accordingly.
> 
> In summary, this tutorial is outdated and no longer supports the current LUIS portal.
> 
> I recommend you follow issue #51 to be notified if/when the tutorial is updated.
> __
> 
> CC @hferrone

@martinwilter 
Thank you. I appreciate the reply.

I will have to take a look at trying to unpack the results and properly recreate the object as required.

Cheers!

**Comment by hferrone** [comment]
@mr0ng Can you include this fix in the tutorials update please?
